src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -1260,7 +1260,7 @@ private SynthesizedRecordConstructor TryGetSynthesizedRecordConstructor(RecordDe
             NamedTypeSymbol recordType = GetDeclaredType(node);
             var symbol = recordType.GetMembersUnordered().OfType<SynthesizedRecordConstructor>().SingleOrDefault();
 
-            if (symbol?.GetSyntax() != node)
+            if (symbol?.SyntaxRef.SyntaxTree != node.SyntaxTree || symbol.GetSyntax() != node)
```

Comment:
Can any behavioral change be demonstrated by a test?

Reply:
> Can any behavioral change be demonstrated by a test?

Theoretically, yes. But I don't think it is worth the effort and we usually don't add tests like that.